KaleidoscopeLang/Expression.swift

```diff
@@ -21,7 +21,7 @@ extension Expression where Self : Equatable {
         return false
     }
 }
-public func == (left: Expression, right: Expression) -> Bool {
+public func == <E: Expression> (left: E, right: E) -> Bool {
```

Comment:
Interesting… where were you seeing an infinite loop? It’s working as-is for me on my slightly outdated Swift 2.1 (Xcode 7.1). Wouldn’t be surprised to hear that this behavior changed *ever* so slightly somewhere along the way, though.